addons/website_slides_survey/controllers/slides.py

```diff
@@ -23,29 +23,38 @@ def _get_slide_detail(self, slide):
         The user still has to have the necessary rights, please check survey.survey._check_answer_creation
         for more info.
         This is used in the context of a website_publisher designing a course."""
-        result = super(WebsiteSlides, self)._get_slide_detail(slide)
+        result = super(WebsiteSlides, self)._get_slide_detail(slide, is_fullscreen)
         if not request.env.user._is_public() and slide.slide_type == 'certification' and slide.survey_id:
-            result['certification_test_entry'] = not slide.channel_id.is_member
-            result['certification_done'] = False
-            if slide.channel_id.is_member:
-                user_membership_id_sudo = slide.user_membership_id.sudo()
-                quizz_passed = user_membership_id_sudo.survey_quizz_passed
-                result['certification_done'] = quizz_passed
-                result['survey_id'] = slide.survey_id.id
-                if not quizz_passed:
-                    last_user_input = next(user_input for user_input in user_membership_id_sudo.user_input_ids.sorted(
-                        lambda user_input: user_input.create_date, reverse=True
-                    ))
-                    result['certification_url'] = last_user_input._get_survey_url()
+            if not is_fullscreen:
```

Comment:
in any case, you should have a dict in result like this
```
resutl = {
slide: ....
next_slide: ...
certification_data: {
    <current_slide_id>: {
         certification_url: ... survey_id: ..., certification_done: ...
    }
   <other_slide_id>: {
          certification_url: ... survey_id: ..., certification_done: ...
   }
}
```
So set `certification_data` to a empty dict. After, check if slide type is certif, then add its certif data in that dict.
Then, if fullscreen, add certif data of other slide (check in category_data, ....) that are already in `result`.